docs/design/2020-09-28-cluster-read-only-mode.md

```diff
@@ -0,0 +1,233 @@
+# Proposal: Support Global Read Only Mode
+
+- Author(s):    [ichn-hu](https://github.com/ichn-hu)
+- Last updated:  Oct. 09, 2020
+- Discussion at:  
+
+## Abstract
+
+This document describes the design of the `read_only` global variable that once turned on blocks non-super users from updating the TiDB cluster.
+
+
+## Motivation or Background
+
+As requested and described in [this issue](https://github.com/pingcap/tidb/issues/18426), such a variable can be beneficial to maintenance, migration and replication, and according to a [blog post](https://pingcap.com/blog/heterogeneous-database-replication-to-tidb#creating-a-read-only-or-archive-database), TiDB can be used as a read-only or archive database, in which case we would like to make the cluster running in a read only mode, while only allowing CDC tools to update the data.
+
+
+## Detailed Design
+
+Based on the issue description and [MySQL’s description](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_read_only), the `read_only` option of TiDB will act similarly.
+
+
+### Behavior Description
```

Comment:
Need the description when a client calls `set @@global.read_only = on` but lose the connection to the server.

Reply:
updated.